varinfo/cmr_search.py

```diff
@@ -75,29 +81,38 @@ def get_granule_link(granule_response: list) -> str:
     return granule_link
 
 
-def get_granule_name(granule_response: list) -> str:
-    ''' Syntax for extracting the granule name from the granule metadata.
-    '''
-    try:
-        return granule_response[0]['producer_granule_id']
-        # ie. MERRA2_100.tavg1_2d_slv_Nx.19800320.nc4
-    except Exception:
-        raise IndexError('No key named "producer_granule_id"')
-
-
-def download_granule(granule_link: str, out_filename: str):
+def download_granule(granule_link: str = None,
+                     out_directory: str = os.getcwd(),
+                     token: str = None) -> str:
     ''' Use the requests module to download data via https.
+        * granule_link: granule download URL.
+        * out_directory: path to save downloaded granule
+            (the default is the current directory).
+        * token: Earthdata Login (EDL) bearer_token
     '''
-    # Check if a string was entered for input parameters
-    if isinstance(granule_link, str) and isinstance(out_filename, str):
-        f = open(out_filename, 'w+')
+    if granule_link is not None:  # Check if input parameter was entered
+        if isinstance(granule_link, str):  # Check if granule link is a string
+            # Check if `out_directory` is a directory and create out_filename
+            if os.path.isdir(out_directory):
```

Comment:
First up - this is great thinking to catch the case of a non-existent directory.

What would you reckon to trying to create the directory if it doesn't already exist:

```
if not isdir(out_directory):
   <make directory>

out_file = os.path.join(out_directory, os.path.basename(granule_link))
```

Also, the nice thing about `os.path.join` above is that it will use the correct type of slash to concatenate file paths depending on the OS running the code - Windows file paths use `\` instead of `/`.